api/auth_middleware_test.go

```diff
@@ -1,63 +1,101 @@
+		expectedResponseCode    int
+	}{
+		{
+			testName:                "KeyPairAuthorization",
+			serviceAccount:          keyPairSA,
+			requestHeaders:          map[string]string{"authorization": fmt.Sprintf("keypair %s:%s", keyPairSA.KeyID, keyPairSA.KeySecret)},
+			expectedResponseHeaders: map[string]string{"x-service-account-name": "keyPairUser"},
+			expectedResponseCode:    http.StatusOK,
+		},
+		{
+			testName:                "OAuthAuthorization",
+			serviceAccount:          oauthSA,
+			requestHeaders:          map[string]string{"authorization": fmt.Sprintf("bearer %s", token.AccessToken)},
+			expectedResponseHeaders: map[string]string{"x-email": oauthSA.Email},
+			expectedResponseCode:    http.StatusOK,
+		},
+		{
+			testName:                "WrongOAuthTokenAuthorization",
+			serviceAccount:          oauthSA,
+			requestHeaders:          map[string]string{"authorization": fmt.Sprintf("bearer %s", "wrong token")},
+			expectedResponseHeaders: emptyMap,
+			expectedResponseCode:    http.StatusUnauthorized,
+		},
+		{
+			testName:                "WrongKeyPairAuthorization",
+			serviceAccount:          keyPairSA,
+			requestHeaders:          map[string]string{"authorization": fmt.Sprintf("keypair %s:%s", "wrong key_id", "wrong_key_secret")},
+			expectedResponseHeaders: emptyMap,
+			expectedResponseCode:    http.StatusUnauthorized,
+		},
+		{
+			testName:                "UndefinedAuthorization",
+			serviceAccount:          nil,
+			requestHeaders:          emptyMap,
+			expectedResponseHeaders: emptyMap,
+			expectedResponseCode:    http.StatusUnauthorized,
+		},
+		{
+			testName:                "UnsupportedAuthorization",
+			serviceAccount:          oauthSA,
+			requestHeaders:          map[string]string{"authorization": fmt.Sprintf("basic %s", b64.StdEncoding.EncodeToString([]byte("user:password")))},
+			expectedResponseHeaders: emptyMap,
+			expectedResponseCode:    http.StatusUnauthorized,
+		},
 	}
-}
 
-func TestAuthMiddlewareBearerShouldAuthenticateUser(t *testing.T) {
-	helpers.CleanupPG(t)
+	for _, testCase := range testCases {
+		t.Run(testCase.testName, func(t *testing.T) {
+			app := helpers.GetApp(t)
+			req, err := http.NewRequest(http.MethodGet, "/service_accounts", nil)
 
-	rootSA := helpers.CreateRootServiceAccountWithOAuth(t)
-	tokensRepo := helpers.GetRepo(t).Tokens
-	tokens, _ := tokensRepo.FindByEmail(rootSA.Email)
-	token := tokens[0]
-	app := helpers.GetApp(t)
-	req, err := http.NewRequest(http.MethodGet, "/service_accounts", nil)
+			if err != nil {
+				t.Errorf("Could not create HTTP request")
+			}
 
-	if err != nil {
-		t.Errorf("Could not create HTTP request for /service_accounts")
-	}
+			for header, content := range testCase.requestHeaders {
+				req.Header.Set(header, content)
+			}
 
-	req.Header.Set("Authorization", fmt.Sprintf("bearer %s", token))
-	response := helpers.DoRequest(t, req, app.GetRouter())
+			response := helpers.DoRequest(t, req, app.GetRouter())
```

Comment:
FYI, abbreviations and short names are fair play in Go. That doesn't mean you shouldn't use more descriptive names, especially for long-lived/ambiguous variables, but nothing wrong in calling it `resp` if you wanted.

(No need to change anything, just an FYI.)